huggingface_pipelines/audio.py

```diff
@@ -0,0 +1,262 @@
+            fbank_dtype=torch.float16,
+            n_parallel=4,
+            pad_idx=0,
+            audio_column="audio",
+            device="cuda",
+            batch_size=32,
+            columns=["audio", "text"],
+            output_path="/path/to/output",
+            output_column_suffix="embedding"
+        )
+
+    """
+
+    encoder_model: str = "text_sonar_basic_encoder"
+    fbank_dtype: torch.dtype = torch.float32
+    n_parallel: int = 4
+    pad_idx: int = 0
+
+
+class HFAudioToEmbeddingPipeline(Pipeline):
+    """
+    A pipeline for converting audio to embeddings using a HuggingFace model.
+
+    This pipeline processes batches of audio data, converting them to embeddings
+    using a specified encoder model. It handles temporary file creation for audio
+    processing and ensures consistent embedding shapes across the batch.
+
+    Attributes:
+        config (HFAudioToEmbeddingPipelineConfig): The configuration for this pipeline.
+        model (SpeechToEmbeddingModelPipeline): The underlying model used for embedding generation.
+
+    Example:
+
+        pipeline_config = HFAudioToEmbeddingPipelineConfig(
+            encoder_model="sonar_speech_encoder",
+            device="cuda",
+            batch_size=16,
+            audio_column="audio"
+        )
+
+        pipeline = HFAudioToEmbeddingPipeline(pipeline_config)
+    """
+
+    def __init__(self, config: HFAudioToEmbeddingPipelineConfig):
+        """
+        Initialize the HFAudioToEmbeddingPipeline.
+
+        Args:
+            config (HFAudioToEmbeddingPipelineConfig): The configuration for this pipeline.
+        """
+        super().__init__(config)
+        self.config = config
+        self.model = SpeechToEmbeddingModelPipeline(
+            encoder=self.config.encoder_model,
+            device=torch.device(self.config.device),
+            fbank_dtype=self.config.fbank_dtype,
+        )
+
+    def process_batch(self, batch: Dict[str, Any]) -> Dict[str, Any]:
+        """
+        Process a batch of audio data, converting it to embeddings.
+
+        This method handles the conversion of audio data to temporary WAV files,
+        generates embeddings using the model, and ensures consistent embedding
+        shapes across the batch.
+
+        Args:
+            batch (Dict[str, Any]): A dictionary containing the batch data.
+                Expected to have an 'audio' key with a list of audio data dictionaries.
+
+        Returns:
+            Dict[str, Any]: The input batch dictionary with an additional key
+                '{column}_{suffix}' containing the generated embeddings.
+
+        Raises:
+            Exception: If there's an error during batch processing or embedding generation.
+        """
+
+        try:
+            for column in self.config.columns:
+                if column not in batch:
+                    logger.warning(f"Column {column} not found in batch. Skipping.")
+                    continue
```

Comment:
I would raise in this case instead of skipping

Reply:
Done.